micro_sam/evaluation/benchmark_datasets.py

```diff
@@ -0,0 +1,448 @@
+            x0, y0, x1, y1 = crop_box
+            crop_image, crop_gt = image[y0: y1, x0: x1], gt[y0: y1, x0: x1]
+
+            # NOTE: There could be a case where some later patches are invalid.
+            if per_n_instance == 1:
+                break
+
+            image_crops.append(crop_image)
+            gt_crops.append(crop_gt)
+
+            # NOTE: If the number of patches extracted have been fulfiled, we stop sampling patches.
+            if len(image_crops) > 0 and i >= desired_tiles_per_image:
+                break
+
+        # Finally, let's store all the patches
+        for image_crop, gt_crop in zip(image_crops, gt_crops):
+            fname = f"{dataset_choice}_{image_counter:05}.tif"
+            assert image_crop.shape == gt_crop.shape
+            imageio.imwrite(os.path.join(save_image_dir, fname), image_crop, compression="zlib")
+            imageio.imwrite(os.path.join(save_gt_dir, fname), gt_crop, compression="zlib")
+            image_counter += 1
+
+    return save_image_dir, save_gt_dir, ndim
+
+
+def _get_image_label_paths(image_dir, gt_dir):
+    image_paths = natsorted(glob(os.path.join(image_dir, "*")))
+    gt_paths = natsorted(glob(os.path.join(gt_dir, "*")))
+    return image_paths, gt_paths
+
+
+def _run_automatic_segmentation_per_dataset(
+    image_paths: List[Union[os.PathLike, str]],
+    gt_paths: List[Union[os.PathLike, str]],
+    model_type: str,
+    output_folder: Union[os.PathLike, str],
+    device: Optional[Union[torch.device, str]] = None,
+    ndim: Optional[int] = None,
+    checkpoint_path: Union[os.PathLike, str] = None,
+    **auto_seg_kwargs
+):
+    """Functionality to run automatic segmentation for multiple input files at once.
+    It stores the evaluated automatic segmentation results (quantitative).
+
+    Args:
+        image_paths:
+        gt_paths:
+        model_type:
+        output_folder,
+        ndim:
+        checkpoint_path:
+        auto_seg_kwargs:
+    """
+    result_path = os.path.join(output_folder, "results", "automatic_segmentation.csv")
+    prediction_dir = os.path.join(output_folder, "automatic_instance_segmentation", "inference")
+    os.makedirs(prediction_dir, exist_ok=True)
+
+    for image_path in tqdm(image_paths, desc="Run automatic segmentation"):
+        output_path = os.path.join(prediction_dir, os.path.basename(image_path))
+        if os.path.exists(output_path):
+            continue
+
+        # Run Automatic Instance Segmentation (AIS)
+        automatic_instance_segmentation(
+            input_path=image_path,
+            output_path=output_path,
+            model_type=model_type,
+            device=device,
+            checkpoint_path=checkpoint_path,
+            ndim=ndim,
+            verbose=False,
+            **auto_seg_kwargs
+        )
+
+    prediction_paths = natsorted(glob(os.path.join(prediction_dir, "*")))
+    res = run_evaluation(gt_paths=gt_paths, prediction_paths=prediction_paths, save_path=result_path)
+    print(res)
```

Comment:
We should create a dataframe and then save it as csv.

Reply:
The results are stored at `result_path` filepath, passed to the `save_path` argument. I'll refactor the evaluation for clarity. Thanks for pointing out!

Reply:
Ok. I think we should also remove the prints of results here at the end. Otherwise the output gets confusing.